cmake/macros/macro_expand_instantiations.cmake

```diff
@@ -78,7 +78,12 @@ MACRO(EXPAND_INSTANTIATIONS _target _inst_in_files)
   #
   FOREACH(_build ${DEAL_II_BUILD_TYPES})
     STRING(TOLOWER ${_build} _build_lowercase)
+
     ADD_DEPENDENCIES(${_target}.${_build_lowercase} ${_target}.inst)
```

Comment:
Interesting. I had always assumed that cmake automatically figures out which .cc file depends on which .inst file and generates it as necessary. 

This way they are generated even earlier. But cmake must still know the dependencies, right? Is the problem that it just can't figure them out for CUDA? If it can't figure them out, how does it know when to recompile a CUDA file if an .inst.in file has changed?